src/screens/qr/components/Toolbar.tsx

```diff
@@ -0,0 +1,61 @@
+import React, {useCallback} from 'react';
+import {StyleSheet} from 'react-native';
+import {Box, Button} from 'components';
+import {useNavigation} from '@react-navigation/native';
+import {useI18n} from 'locale';
+import {BackButton} from '../views/BackButton';
+
+export interface ToolbarProps {
+  onIconClicked(): void;
+  navText?: string;
+  showBackButton?: boolean;
+  showCloseButton?: boolean;
+  isWhite?: boolean;
+}
+
+export const Toolbar = ({navText, onIconClicked, showBackButton, showCloseButton, isWhite}: ToolbarProps) => {
+  return (
+    <Box flexDirection="row" alignItems="center" minHeight={56}>
+      {showBackButton ? <ToolbarWhite isWhite={isWhite} /> : null}
+      {showCloseButton ? (
+        <Box style={styles.right}>
+          <Button testID="toolbarCloseButton" text={navText} variant="text" onPress={onIconClicked} />
+        </Box>
+      ) : null}
+    </Box>
+  );
+};
+
+export const ToolbarWhite = ({isWhite}: {isWhite?: boolean}) => {
```

Comment:
The name `ToolbarWhite` makes me think this is the white toolbar, not that it will switch. Maybe just call it `QRToolbar`? Also the prop could be called `useWhiteText` so it's obvious what is changing colour.